### PR TITLE
Added model function to create and get user future status

### PR DESCRIPTION
**Pull request by sahsisunny**
## Description

- Added model functions to create and get user future status
### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [x] Yes
- [ ] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

File: types/userFutureStatus.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export type UserFutureStatusType = {
+  id?: string;
+  requestId?: string;
+  status: userState.OOO | userState.IDLE | userState.ACTIVE;
+  state: statusState.UPCOMING | statusState.APPLIED | statusState.NOT_APPLIED;
+  from: number;
+  endsOn?: number;
+  userId: string;
+  message?: string;
+  updatedAt?: number;
```
Comment (lakshayman) [inline types/userFutureStatus.d.ts:12]: why there is an updatedAt field?

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: types/userFutureStatus.d.ts (hunk 1 above)
Comment (sahsisunny) [reply types/userFutureStatus.d.ts:12]: When this doc is updated it can be helpful

Comment (lakshayman) [reply types/userFutureStatus.d.ts:12]: According to the implementation, the doc never updates. It recreates a new doc.

File: models/userFutureStatus.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Function to create user future status
+ * @param body: UserFutureStatusType
+ * @returns UserFutureStatusType
+ */
+export const createUserFutureStatus = async (body: UserFutureStatusType) => {
+  try {
+    const statusBody: UserFutureStatusType = {
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
```
Comment (sahsisunny) [inline models/userFutureStatus.ts:15]:
```suggestion
```

File: types/userFutureStatus.d.ts (hunk 1 above)
Comment (sahsisunny) [inline types/userFutureStatus.d.ts:12]:
```suggestion
```

Comment (sahsisunny) [reply types/userFutureStatus.d.ts]: Removed

**Comment by lakshayman** [comment]
Test failing

**Review by lakshayman — APPROVED** [review]

**Pull request merged**